src/overlays/actors/ovl_Bg_Last_Bwall/z_bg_last_bwall.c

```diff
@@ -28,57 +49,217 @@ const ActorInit Bg_Last_Bwall_InitVars = {
     {
-        { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+        {
+            ELEMTYPE_UNK0,
+            { 0x00000000, 0x00, 0x00 },
+            { 0x00000008, 0x00, 0x00 },
+            TOUCH_NONE | TOUCH_SFX_NORMAL,
+            BUMP_ON,
+            OCELEM_NONE,
+        },
         { { { 70.0f, 0.0f, 3.0f }, { 70.0f, 200.0f, 3.0f }, { -70.0f, 200.0f, 3.0f } } },
     },
 };
 
-// static ColliderTrisInit sTrisInit = {
-static ColliderTrisInit D_80C18A38 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_TRIS, },
-    ARRAY_COUNT(sTrisElementsInit), D_80C189C0, // sTrisElementsInit,
+static ColliderTrisInit sTrisInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_TRIS,
+    },
+    ARRAY_COUNT(sTrisElementsInit),
+    sTrisElementsInit, // sTrisElementsInit,
+};
+
+static BgLastBwallModelInfo D_80C18A48[] = {
+    { object_last_obj_DL_0001A8, NULL, &object_last_obj_Colheader_000288 },
+    { object_last_obj_DL_001318, NULL, &object_last_obj_Colheader_001450 },
+};
+
+static Vec3s D_80C18A60[] = {
+    { 0x118, 0x118, -0x28 },
+    { 0x118, 0x118, 0x78 },
+    { 0x118, 0x190, 0x78 },
+    { 0x118, 0x190, -0x28 },
+};
+
+static s16 D_80C18A78[] = { 0, 1, 2, 0, 2, 3 };
+
+static BgLastBwallInitColliderStruct D_80C18A84 = { D_80C18A60, D_80C18A78, 2 };
+
+static Vec3s D_80C18A90[] = {
+    { -0x50, 0x190, -0x50 },
+    { -0x50, 0x190, 0x50 },
+    { 0x50, 0x190, 0x50 },
+    { 0x50, 0x190, -0x50 },
 };
 
-// static InitChainEntry sInitChain[] = {
+static s16 D_80C18AA8[] = { 0, 3, 2, 0, 2, 1 };
+
+static BgLastBwallInitColliderStruct D_80C18AB4 = { D_80C18A90, D_80C18AA8, 2 };
+
+static BgLastBwallInitColliderStruct* D_80C18AC0[] = { &D_80C18A84, &D_80C18AB4 };
+
 static InitChainEntry D_80C18AC8[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgLastBwall_InitCollider(ColliderTrisInit* init, Vec3f* pos, Vec3s* rot, ColliderTris* collider,
+                              BgLastBwallInitColliderStruct* arg4) {
+    s32 i;
+    s32 j;
+    Vec3f sp6C[3];
+    Vec3f sp60;
+    Vec3f sp54;
+
+    Matrix_RotateYS(rot->y, MTXMODE_NEW);
+    Matrix_RotateXS(rot->x, MTXMODE_APPLY);
+    Matrix_RotateZS(rot->z, MTXMODE_APPLY);
+    for (i = 0; i < init->count; i++) {
+        for (j = 0; j < 3; j++) {
+            Math_Vec3s_ToVec3f(&sp60, &arg4->unk0[arg4->unk4[j]]);
+            Matrix_MultVec3f(&sp60, &sp54);
+            Math_Vec3f_Sum(&sp54, pos, &sp6C[j]);
+        }
+
+        arg4->unk4 += 3;
+        Collider_SetTrisVertices(collider, i, &sp6C[0], &sp6C[1], &sp6C[2]);
+    }
+}
+
+void BgLastBwall_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    BgLastBwall* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, D_80C18AC8);
+    this->unk238 = BGLASTBWALL_GET_F000(&this->dyna.actor);
```

Comment:
Looks like this is the type, please make an enum for the two different values it uses.